Test notebooks

billbrod opened the pull request:
Adds integration with treebeard (with corresponding badge) so that we run all notebooks every push. Similar to our testing set up on Travis CI, this makes uses of a build matrix to test pythons 3.6 and 3.7, as well as doing each file in parallel. The downside of doing them in parallel is that we need to remember to update the corresponding file (`.github/workflows/treebeard.yml`) whenever a new tutorial is added, but it does speed it up.

As part of this:
 - tracked down the weird `tqdm` notebook issue in the Eigendistortions notebook. the `workflows/treebeard.yml` file contains a fix: just make sure ipywidgets are installed
 - removes some old stuff from the perceptual distances notebook
 - linear approximation notebook used a model we no longer have (`Steerable_GainControl`), so switched that to regular steerable pyramid.
 - small changes to steerable pyramid notebook to make sure that worked
 - Original MAD notebook now explicitly downloads the required files (using the helper function from `test_plenoptic.py`)

NOTE: this still fails (probably, it's still running as of right now) because it times out while downloading VGG in the Eigendistortion notebook (all others complete without a problem). Not sure why, it downloads in ~1 minute on my local laptop, but I guess the download speed isn't so good. The treebeard devs are going to onboard me sometime next week for how to view the actual notebook that was run by treebeard, which should give me a better sense for how long it will take. May need to reconsider current use, though, and either come up with another way to get it downloaded or a way to circumvent it for the tests.

Addresses #37

lyndond commented on examples/09_Original_MAD.ipynb:
```diff
@@ -69,15 +68,15 @@
      "output_type": "stream",
      "text": [
-      "/home/billbrod/Documents/plenoptic/plenoptic/metric/perceptual_distance.py:92: UserWarning: dynamic_range is 1 but image range falls outside [0, 1] img1: tensor([0.0039, 1.0000]), img2: tensor([-11.9009,  15.4622]). Continuing anyway...\n",
+      "/home/billbrod/Documents/plenoptic/plenoptic/metric/perceptual_distance.py:92: UserWarning: dynamic_range is 1 but image range falls outside [0, 1] img1: tensor([0.0039, 1.0000]), img2: tensor([-10.7873,  12.8881]). Continuing anyway...\n",
```
Maybe mention this is a work in progress? 
Add text/headers etc to give user an idea of what's going on.

lyndond reviewed (changes requested):
Minor qualms that I left comments for in all but two files.

03_steerable and 04_perceptual are too big for me to add comments in-browser. So I'll write comments here w/ request for changes.

03_steerable:

- I know this one is likely to change after we move to torch.complex. But the changes probably won't affect this nb that much. 
- The content is good overall but could use some better formatting w. headers/subheaders etc to break it up a bit. 
- It says you cannot use odd-size images but you in fact can, it just gives a warning, right?


04_perceptual:
- What is the end-user supposed to take away from this exactly?
- maybe add that this is work in progress

billbrod commented on README.md:
```diff
@@ -201,6 +202,14 @@ extra, but if you closed your terminal session after the last section
 (for example), you'll need to make sure to activate the correct
 environment first: `conda activate plenoptic`.
 
+Note that you may also run into an error in the `02_Eigendistortions` notebook
```
Yes, you're right about that. However, we can get rid of this possibility by setting `progress=False` in the download call. Should we do that? Even though the progress bar is nice and informative.

billbrod commented:
I have switched over our tests to now use Github Actions instead of Travis, so recent changes reflect that (updating badges, CONTRIBUTING).

@lyndond I also added more text to the Original MAD notebook, let me know what you think. The other suggested changes are in notebooks by @pehf and @nikparth, so I'll wait for them to weight in on them.

nikparth commented:
> I have switched over our tests to now use Github Actions instead of Travis, so recent changes reflect that (updating badges, CONTRIBUTING).
> 
> @lyndond I also added more text to the Original MAD notebook, let me know what you think. The other suggested changes are in notebooks by @pehf and @nikparth, so I'll wait for them to weight in on them.

Looks fine to me for the notebook and took a quick look at the rest just in case and didn't see anything glaring. so good to go for me!

billbrod commented:
> > I have switched over our tests to now use Github Actions instead of Travis, so recent changes reflect that (updating badges, CONTRIBUTING).
> > @lyndond I also added more text to the Original MAD notebook, let me know what you think. The other suggested changes are in notebooks by @pehf and @nikparth, so I'll wait for them to weight in on them.
> 
> Looks fine to me for the notebook and took a quick look at the rest just in case and didn't see anything glaring. so good to go for me!

Did you see Lyndon's comments about the notebook?

nikparth commented:
> > > I have switched over our tests to now use Github Actions instead of Travis, so recent changes reflect that (updating badges, CONTRIBUTING).
> > > @lyndond I also added more text to the Original MAD notebook, let me know what you think. The other suggested changes are in notebooks by @pehf and @nikparth, so I'll wait for them to weight in on them.
> > 
> > 
> > Looks fine to me for the notebook and took a quick look at the rest just in case and didn't see anything glaring. so good to go for me!
> 
> Did you see Lyndon's comments about the 03 Steerable Pyramid notebook?

yeah I mean I think given the fact that the notebook is more of a placeholder than anything I don't think we should worry about doing anything with it for now -- after I make edits for the pyramid changes and decide what we really want the notebook to be I think we can look more at those details

nikparth commented:
> Minor qualms that I left comments for in all but two files.
> 
> 03_steerable and 04_perceptual are too big for me to add comments in-browser. So I'll write comments here w/ request for changes.
> 
> 03_steerable:
> 
> * I know this one is likely to change after we move to torch.complex. But the changes probably won't affect this nb that much.
> * The content is good overall but could use some better formatting w. headers/subheaders etc to break it up a bit.
> * It says you cannot use odd-size images but you in fact can, it just gives a warning, right?
> 
> 04_perceptual:
> 
> * What is the end-user supposed to take away from this exactly?
> * maybe add that this is work in progress

I dont see these as changes that we should worry about now- the notebook is probably going to change quite a bit. The odd-size image thing I'll have to check but might be worth changing it to an actual error because the results with the odd-sized images are not really trustworthy if I remember. Regardless this would be something I need to look at first in the pyramid and decided what to do there and then I'll worry about the notebook.
